### PR TITLE
Test subgroup enforcement, improve ElGamal doc

### PR DESCRIPTION
**Pull request by andrew-fleming**
Address one point in #633:

> This identity (ecNeg(P) = (ℓ-1)·P = -P) holds only for prime-order-subgroup points, which rests on the module's Subgroup membership note: every assigned JubjubPoint is cofactor-cleared into the subgroup. Is that confirmed with the compactc / ledger team, or inferred from observed behaviour? Since compactc and midnight-ledger are unaudited, it is the most load-bearing assumption in the module. If it is an assumption rather than a guarantee, consider promoting it from @dev prose to a named trust assumption.

## Summary by CodeRabbit

- **Documentation**
  - Clarified curve-operation safety notes with a stronger trust-assumption statement and more explicit guidance on subgroup handling.
  - Updated developer-facing notes to better explain edge cases in point negation.

- **Tests**
  - Added regression coverage for elliptic-curve operations to verify correct behavior for valid points and failures for invalid, low-order, and mixed-order points.
  - Added test-only helpers to exercise curve operations directly in a controlled environment.

### CODE REVIEW

File: contracts/src/crypto/ElGamal.compact [hunk 1]
```diff
@@ -34,16 +34,29 @@ pragma language_version >= 0.23.0;
...
  *
- * @dev Subgroup membership. Every `JubjubPoint` reaching a curve operation is
- * in the Jubjub prime-order subgroup, including points supplied by a caller as
- * a witness or argument (e.g. a recipient `pk`, or the message point in
- * `encryptPoint`). The Midnight runtime enforces this at the ZK-constraint
+ * @dev TRUST ASSUMPTION — subgroup membership. This is the module's most
+ * load-bearing assumption. Every `JubjubPoint` reaching a curve operation is
+ * assumed to be in the Jubjub prime-order subgroup, including points supplied by
+ * a caller as a witness or argument (e.g. a recipient `pk`, or the message point
+ * in `encryptPoint`). The Midnight runtime enforces this at the ZK-constraint
```
Comment (0xisk) [inline contracts/src/crypto/ElGamal.compact:41]: 👍

File: contracts/src/crypto/ElGamal.compact [hunk 2]
```diff
@@ -34,16 +34,29 @@ pragma language_version >= 0.23.0;
...
+ * exceeds the valid scalar range, while `ℓ-1` is accepted). The `ecNeg` identity
+ * (`(ℓ-1)*P = -P`) and the homomorphic algebra depend on this holding.
+ *
+ * Basis (and its limits): source-verified against `midnight-circuits` 6.0.0 —
+ * the version vendored by the Midnight onchain-runtime we compile against —
+ * specifically `ecc::native::edwards_chip`, where `assign` cofactor-clears the
+ * point and `q_mem`'s `create_membership_gate` enforces on-curve, reached from
+ * the ZKIR `ec_mul`/`ec_add` lowering via `ecc_from_parts`. This is NOT a
+ * team-confirmed or audited guarantee: compactc and midnight-ledger are
+ * unaudited, so treat it as a substantiated-but-unaudited assumption. It is
```
Comment (0xisk) [inline contracts/src/crypto/ElGamal.compact:56]: 👍

**Review by 0xisk — APPROVED** [review]
Thank you @andrew-fleming! Looks good.
